### PR TITLE
DAOS-10029 test: Only call list_tests.py if it exists.

### PR DESCRIPTION
**Pull request by ashleypittman**
If not then call launch.py --list instead.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: vars/testsInStage.groovy [hunk 1]
```diff
@@ -25,6 +25,11 @@ boolean call() {
                              exit 0
                          fi
                          cd src/tests/ftest
-                         ./list_tests.py """ + parseStageInfo()['test_tag'],
+                         if [ -f list_tests.py ]
```
Comment (brianjmurrell) [inline vars/testsInStage.groovy:28]:
```suggestion
                         if [ -e list_tests.py ]
```

It has to be executable, not merely exist, to be run without specifying an interpreter on the command line.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Comment by ashleypittman** [comment]
This works for calling launch.py if list_tests.py doesn't exist.  I'll wait for that run tomorrow and try and verify it for cases where it does.
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8911/8/execution/node/1042/log/

**Comment by ashleypittman** [comment]
And this works for the existing master setup as well where list_tests.py exists.

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8987/1/execution/node/1050/log/

File: vars/testsInStage.groovy (hunk 1 above)
Comment (ashleypittman) [reply vars/testsInStage.groovy]: Done, all working.

File: vars/testsInStage.groovy [hunk 2]
```diff
@@ -25,6 +25,11 @@ boolean call() {
                              exit 0
                          fi
                          cd src/tests/ftest
-                         ./list_tests.py """ + parseStageInfo()['test_tag'],
+                         if [ -x list_tests.py ]
```
Comment (brianjmurrell) [inline vars/testsInStage.groovy:28]: Yeah, oops on my part.  I suggested `-e` but I really did mean `-x`.  Thanks for picking up on that.

File: vars/testsInStage.groovy [hunk 3]
```diff
@@ -25,6 +25,11 @@ boolean call() {
...
+                         then
+                             ./list_tests.py """ + parseStageInfo()['test_tag'] + """
+                         else
+                             ./launch.py --list """ + parseStageInfo()['test_tag'] + """
+                         fi""",
```
Comment (brianjmurrell) [inline vars/testsInStage.groovy:33]: The only other suggestion I would have made last time was a bit of DRYing such as:

```suggestion
                             cmd=('./list_tests.py')
                         else
                             cmd=('./launch.py' '--list')
                         fi
                         "${cmd[@]}"""" + parseStageInfo()['test_tag']
```

**Review by brianjmurrell — APPROVED** [review]

File: vars/testsInStage.groovy (hunk 3 above)
Comment (ashleypittman) [reply vars/testsInStage.groovy:33]: My groovy isn't that good!

Comment (brianjmurrell) [reply vars/testsInStage.groovy:33]: Well, the optimization is actually in the shell script inside the groovy string, so no advanced groovy necessary.  It's all good, in any case.

**Pull request merged**